File: XDS.cpp
#include<iostream>
using namespace std;
#include<math.h>

int main()
{
	int N;
	cin>>N;
	for(int i=0;i<N;i++)
	{
		int n;
		cin>>n;
		bool b=false;

		for(int j=2;j<=sqrt(n);j++)
		{
			if(n%j==0)
				b=true;
		}
	
	if(b)
	{
		int y=(int)floor(sqrt(n));
		int x=n/y;
		for(int j=0;j<x;j++)
			{
				cout<<"X";
			}
			for(int j=0;j<y;j++)
			{
				cout<<"D";
			}

	}
	else
		{
			cout<<"XD";
			if(n>=3)
				{
					cout<<"X";
				}
			for(int j=0;j<floor(n/2);j++)
			{
				
				cout<<"D";
			}	
		}
		cout<<endl;
}
}

File: XDS.java
import java.util.Scanner;

public class Main{
public static void main(String args[])
{
	Scanner s=new Scanner(System.in);
	int N=s.nextInt();
	for(int i=0;i<N;i++)
	{
		int n=s.nextInt();
		boolean b=false;//Number is not Composite

		for(int j=2;j<=Math.sqrt(n);j++)
		{
			if(n%j==0){	//Number is composite
				b=true;
				// break;
			}
		}

		if(b)
		{
			int y=(int)Math.floor(Math.sqrt(n));
			int x=n/y;

			for(int j=0;j<x;j++)
			{
				System.out.print("X");
			}
			for(int j=0;j<y;j++)
			{
				System.out.print("D");
			}
		}
		else
		{
			System.out.print("XD");
			if(n>=3)
				{
					System.out.print("X");
				}
			for(int j=0;j<Math.floor(n/2);j++)
			{
				
				System.out.print("D");
			}	
		}
		System.out.println();
	}
}
}
//XDDDDD